From ab76fb79c2e385be8ec9df2a95fdb5031b02836d Mon Sep 17 00:00:00 2001
From: Praveenkumar I <quic_ipkumar@quicinc.com>
Date: Thu, 21 Nov 2024 17:14:37 +0530
Subject: [PATCH] pcie: qcom: Register the Global IRQ as Level high

IPQ5424 has a new feature as APSS_SHARED_SPI_CONFIG_n to
configure the interrupt as Edge / Level. Global irq is
configured as Level in the HW. Hence changing the Global
IRQ to Level high.

The change works good for all other IPQ targets.

Change-Id: Ideb2b84bac79ff3034b48c4400590e59947c6431
Signed-off-by: Praveenkumar I <quic_ipkumar@quicinc.com>
---
 drivers/pci/controller/dwc/pcie-qcom.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/controller/dwc/pcie-qcom.c b/drivers/pci/controller/dwc/pcie-qcom.c
index 35d69d652977..7c7ec269a063 100644
--- a/drivers/pci/controller/dwc/pcie-qcom.c
+++ b/drivers/pci/controller/dwc/pcie-qcom.c
@@ -1862,7 +1862,7 @@ static int qcom_pcie_probe(struct platform_device *pdev)
 		ret = devm_request_threaded_irq(&pdev->dev, pcie->global_irq,
 					NULL,
 					qcom_pcie_global_irq_thread_fn,
-					IRQF_TRIGGER_RISING | IRQF_ONESHOT,
+					IRQF_TRIGGER_HIGH | IRQF_ONESHOT,
 					"pcie-global", pcie);
 		if (ret) {
 			dev_err(&pdev->dev, "Unable to request global irq\n");
-- 
2.34.1

